aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsMetricAttributeGenerator.java

```diff
@@ -152,35 +212,132 @@ private static void setEgressOperation(SpanData span, AttributesBuilder builder)
-    } else if (isKeyPresent(span, FAAS_INVOKED_PROVIDER) || isKeyPresent(span, FAAS_INVOKED_NAME)) {
-      setRemoteService(span, builder, FAAS_INVOKED_PROVIDER);
-      setRemoteOperation(span, builder, FAAS_INVOKED_NAME);
+      remoteService = getRemoteService(span, DB_SYSTEM);
+      remoteOperation = getRemoteOperation(span, DB_OPERATION);
+    } else if (isKeyPresent(span, FAAS_INVOKED_NAME) || isKeyPresent(span, FAAS_TRIGGER)) {
+      remoteService = getRemoteService(span, FAAS_INVOKED_NAME);
+      remoteOperation = getRemoteOperation(span, FAAS_TRIGGER);
     } else if (isKeyPresent(span, MESSAGING_SYSTEM) || isKeyPresent(span, MESSAGING_OPERATION)) {
-      setRemoteService(span, builder, MESSAGING_SYSTEM);
-      setRemoteOperation(span, builder, MESSAGING_OPERATION);
+      remoteService = getRemoteService(span, MESSAGING_SYSTEM);
+      remoteOperation = getRemoteOperation(span, MESSAGING_OPERATION);
     } else if (isKeyPresent(span, GRAPHQL_OPERATION_TYPE)) {
-      builder.put(AWS_REMOTE_SERVICE, GRAPHQL);
-      setRemoteOperation(span, builder, GRAPHQL_OPERATION_TYPE);
-    } else {
-      logUnknownAttribute(AWS_REMOTE_SERVICE, span);
-      builder.put(AWS_REMOTE_SERVICE, UNKNOWN_REMOTE_SERVICE);
-      logUnknownAttribute(AWS_REMOTE_OPERATION, span);
-      builder.put(AWS_REMOTE_OPERATION, UNKNOWN_REMOTE_OPERATION);
+      remoteService = GRAPHQL;
+      remoteOperation = getRemoteOperation(span, GRAPHQL_OPERATION_TYPE);
     }
 
     // Peer service takes priority as RemoteService over everything but AWS Remote.
     if (isKeyPresent(span, PEER_SERVICE) && !isKeyPresent(span, AWS_REMOTE_SERVICE)) {
-      setRemoteService(span, builder, PEER_SERVICE);
+      remoteService = getRemoteService(span, PEER_SERVICE);
     }
+
+    // try to derive RemoteService and RemoteOperation from the other un-directive attributes
+    if (remoteService.equals(UNKNOWN_REMOTE_SERVICE)) {
+      remoteService = generateRemoteService(span);
+    }
+    if (remoteOperation.equals(UNKNOWN_REMOTE_OPERATION)) {
+      remoteOperation = generateRemoteOperation(span);
+    }
+
+    builder.put(AWS_REMOTE_SERVICE, remoteService);
+    builder.put(AWS_REMOTE_OPERATION, remoteOperation);
+  }
+
+  /**
+   * When span name is not meaningful(null, unknown or http_method value) as operation name for http
+   * use cases. Will try to extract the operation name from http target string
+   */
+  private static String generateIngressOperation(SpanData span) {
+    String operation = UNKNOWN_OPERATION;
+    if (isKeyPresent(span, HTTP_TARGET)) {
+      String httpTarget = span.getAttributes().get(HTTP_TARGET);
+      // get the first part from API path string as operation value
+      // the more levels/parts we get from API path the higher chance for getting high cardinality
+      // data
+      if (httpTarget != null) {
+        operation = extractAPIPathValue(httpTarget);
+      }
+      if (isKeyPresent(span, HTTP_METHOD)) {
```

Comment:
if `HTTP_METHOD` is not found, are we sure we want to keep just the first part of the target?

Also this code is confusing because in `isValidOperation`, we are saying that HTTP_METHOD alone is invalid, but here we consider it valid if target is null. A cleaner solution would be to update the if block on line 158 to attempt to append target if only HTTP_METHOD is being used as span name.

Reply:
changed it to: only add `http.method` as prefix for RemoteOperation when `http.target` is not null.